torchmultimodal/modules/layers/attention.py

```diff
@@ -102,15 +100,18 @@ class MultiHeadAttention(nn.Module):
 
     def __init__(
         self,
-        shape: Tuple[int, ...],
         dim_q: int,
         dim_kv: int,
         n_head: int,
         n_layer: int,
         causal: bool,
         attn_module: nn.Module,
+        shape: Optional[Tuple[int]] = None,
     ) -> None:
         super().__init__()
+        if causal and shape is None:
```

Comment:
and other way round also right is causal is false, shape should be none

Reply:
shape is ignored if causal is false, so it should be ok as is